staging/src/kubevirt.io/client-go/kubecli/vm.go

```diff
@@ -228,6 +228,23 @@ func (v *vm) Migrate(name string, migrateOptions *v1.MigrateOptions) error {
 	return v.restClient.Put().RequestURI(uri).Body(optsJson).Do(context.Background()).Error()
 }
 
+func (v *vm) MemoryDump(name string, memoryDumpRequest *v1.VirtualMachineMemoryDumpRequest) error {
```

Comment:
why is this on VM and not VMI?

Reply:
hmm it is true that the memory dump is basically an action that happens on the vmi and there is a check that the vmi exists and running, but the action starts by updating the vm since eventually the memory dump info will be only on the vm and not on the vmi since the output does outlives the vmi. also the counteraction of memory dump - remove-memory-dump is an action solely on the vm and can happen on the vm even if vmi doesnt exist.
You think the subresource should be on the vmi and the function that is being called will update the vm status and not the vmi?

Reply:
Maybe both VM and VMI should have memorydump endpoints?

Reply:
which both do the same thing of updating the vm status with the memory dump request?

Reply:
If you refer for doing memory dump without the whole part of updating the VM I guess its possible but then the volume will still be mounted to the VMI. and anyways maybe I can add it later as an improvement not in this PR.